hazelcast/src/main/java/com/hazelcast/util/ServiceLoader.java

```diff
@@ -116,9 +113,8 @@ private ServiceLoader() {
                 URL url = configs.nextElement();
                 URI uri = new URI(url.toExternalForm().replace(" ", "%20"));
 
-                ClassLoader highestClassLoader = findHighestReachableClassLoader(url, classLoader, resourceName);
-                if (!highestClassLoader.getClass().getName().equals(IGNORED_GLASSFISH_MAGIC_CLASSLOADER)) {
-                    urlDefinitions.add(new URLDefinition(uri, highestClassLoader));
+                if (!classLoader.getClass().getName().equals(IGNORED_GLASSFISH_MAGIC_CLASSLOADER)) {
```

Comment:
IntelliJ is warning me this line can cause NPE. I believe it can, since the `ArrayList` can contain `null` and it could be added to the list as a result of this:
```
        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
        if (tccl != classLoader) {
            classLoaders.add(tccl);
        }
```
Where `getContextClassLoader` can return `null`:
```
     * @return  the context ClassLoader for this Thread, or {@code null}
     *          indicating the system class loader (or, failing that, the
     *          bootstrap class loader)
```